include/arch/sparcv8leon3.h

```diff
@@ -0,0 +1,203 @@
+
+/* Serial communication */
+#define CORE_ID_AHBUART     0x007u
+#define CORE_ID_AHBJTAG     0x01cu
+#define CORE_ID_APBPS2      0x060u
+#define CORE_ID_APBUART     0x00cu
+#define CORE_ID_CAN_OC      0x019u
+#define CORE_ID_CANMUX      0x081u
+#define CORE_ID_GRCAN       0x03du
+#define CORE_ID_GRCANFD     0x0b5u
+#define CORE_ID_GRHSSL      0x0c8u
+#define CORE_ID_GRSPFI      0x0bcu
+#define CORE_ID_GRSPW2      0x029u
+#define CORE_ID_GRSPWROUTER 0x03eu
+#define CORE_ID_GRSPWTDP    0x097u
+#define CORE_ID_GRSRIO      0x0a8u
+#define CORE_ID_GRWIZL      0x0c7u
+#define CORE_ID_I2C2AHB     0x00bu
+#define CORE_ID_I2CMST      0x028u
+#define CORE_ID_I2CSLV      0x03eu
+#define CORE_ID_SOCBRIDGE   0x0c4u
+#define CORE_ID_SPI2AHB     0x05cu
+#define CORE_ID_SPICTRL     0x02du
+#define CORE_ID_SPIMASTER   0x0a6u
+#define CORE_ID_SPISLAVE    0x0a7u
+
+/* Ethernet interface */
+#define CORE_ID_GRETH      0x01du
+#define CORE_ID_GRETH_GBIT 0x01du
+#define CORE_ID_RGMII      0x093u
+
+/* USB interface */
+#define CORE_ID_GRUSBHC 0x027u
+#define CORE_ID_GRUSBDC 0x022u
+
+/* MIL-STD-1553B interface */
+#define CORE_ID_GR1553B    0x04du
+#define CORE_ID_GRB1553BRM 0x072u
+
+/* Encryption and compression */
+#define CORE_ID_GRAES     0x073u
+#define CORE_ID_GRAES_DMA 0x07bu
+#define CORE_ID_GRECC     0x074u
+#define CORE_ID_GRSHYLOC  0x0b7u
+
+/* FPGA control and error mitigation */
+#define CORE_ID_GRSCRUB 0x0c1u
+
+/* Graphics functions */
+#define CORE_ID_APBVGA   0x061u
+#define CORE_ID_SVGACTRL 0x063u
+
+/* Auxiliary functions */
+#define CORE_ID_GRACECTRL 0x067u
+#define CORE_ID_GRADCDAC  0x036u
+#define CORE_ID_GRFIFO    0x035u
+#define CORE_ID_GRGPIO    0x01au
+#define CORE_ID_GRGPREG   0x087u
+#define CORE_ID_GRGPRBANK 0x08fu
+#define CORE_ID_GRPULSE   0x037u
+#define CORE_ID_GRPWM     0x04au
+#define CORE_ID_GRSYSMON  0x066u
+#define CORE_ID_GRVERSION 0x03au
+
+/* Spacecraft data handling functions */
+#define CORE_ID_GRTM      0x030u
+#define CORE_ID_GRTM_DESC 0x084u
+#define CORE_ID_GRTM_VC   0x085u
+#define CORE_ID_GRTM_PAHB 0x088u
+#define CORE_ID_GRGEFFE   0x086u
+#define CORE_ID_GRTMRX    0x082u
+#define CORE_ID_GRTC      0x031u
+#define CORE_ID_GRTCTX    0x083u
+#define CORE_ID_GRCTM     0x033u
+#define CORE_ID_SPWCUC    0x089u
+#define CORE_ID_GRPW      0x032u
+#define CORE_ID_GRPWRX    0x08eu
+#define CORE_ID_GRPWTX    0x08du
+#define CORE_ID_APB2PW    0x03bu
+#define CORE_ID_PW2APB    0x03cu
+#define CORE_ID_AHB2PP    0x039u
+#define CORE_ID_GRRM      0x09au
+
+#define AMBA_TYPE_APBIO  0x1u
+#define AMBA_TYPE_AHBMEM 0x2u
+#define AMBA_TYPE_AHBIO  0x3u
+
+
+typedef struct _ambapp_dev_t {
+	unsigned short devId;
+	unsigned char vendor;
+	unsigned char irqn;
+	/* clang-format off */
+	enum { amba_ahb = 0, amba_apb } bus;
+	/* clang-format on */
+	union {
+		struct {
+			unsigned int *base[4];
+			unsigned int type[4];
+		} ahbInfo;
+		struct {
+			unsigned int *base;
+			unsigned int type;
+		} apbInfo;
+	};
```

Comment:
Anonymous union within struct is a gnu extension (and part of C11). I know we have many examples, but they'll have to go eventually. Perhaps name union `info` and it's elements `ahb` and `apb`?